### ISSUE TITLE
Spike: client side CSP unsafe inline styles resolution

### ISSUE THREAD

**Issue by jorgegonzalez** [raft-tech/TANF-app#1208]
**Description:**

Per #907 we added a client-side Content Security Policy header with the PR #1142. This resolved the missing CSP headers warning from the frontend OWASP report, but unfortunately introduced a new OWASP warning.

![image](https://user-images.githubusercontent.com/12901172/128537144-7747bc19-0f92-495d-9c40-1061a7b77d45.png)

We require this header because of the inline styles generated from our dependencies on `uswds` and `node-sass`. The work to address this problem was out of scope for #907 and will require something like a custom build process outside of `react-scripts` to add a dynamically generated nonce/sha256 to the static bundle and the nginx configurations. More research is needed to see if this is viable and solves the problem.

**References:**
- https://stackoverflow.com/questions/30653698/csp-style-src-unsafe-inline-is-it-worth-it
- https://content-security-policy.com/style-src/
- https://cssinjs.org/csp/?v=v10.7.1

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] _Outcome 1_
- [ ] _Outcome 2_
- [ ] _Outcome 3_
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] _Task 1_
- [ ] _Task 2_
- [ ] _Task 3_
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
_Add additional useful information, such as related issues and functionality that isn't covered by this specific issue, and other considerations that will be helpful for anyone reading this_ 

- _Note 1_
- _Note 2_
- _Note 3_

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_

- _Doc 1_
- _Doc 2_

**Open Questions:**
_Please include any questions or decisions that must be made before beginning work or to confidently call this issue complete_ 

- _Open Question 1_
- _Open Question 2_

**Comment by amilash** [comment]
@jorgegonzalez  is this blocking work that we need to do sooner than later? Asking to know where to prioritize this in the coming sprints/versions. Thanks!

**Comment by jorgegonzalez** [comment]
This is not necessarily blocking other work, but it will be needed after #1142 is merged.

**Comment by abottoms-coder** [comment]
https://medium.com/@nrshahri/csp-cra-324dd83fe5ff

**Comment by jorgegonzalez** [comment]
FYI good examples here https://developers.google.com/web/fundamentals/security/csp/#inline_code_is_considered_harmful

**Comment by abottoms-coder** [comment]
Tasks:
- [ ] Contact 18F, the maintainers of uswds, about this issue
- [ ] Investigate React-app-rewired
- [ ] If neither of the above resolve, will need to waive this or migrate to a server-side rendering framework ala next.js

**Comment by abottoms-coder** [comment]
Let's timebox further research on this just until end of week.
